spec/models/tag_spec.rb

```diff
@@ -122,6 +122,18 @@
     end
   end
 
+  it "triggers articles cache busting on save" do
+    sidekiq_perform_enqueued_jobs do
+      tag.save
+    end
+    create(:article, tags: tag.name, published: true)
+    create(:article, tags: tag.name, published: true)
+    article_ids = Article.tagged_with(tag.name).published.order(created_at: :desc).limit(100).ids
+    sidekiq_assert_enqueued_with(job: Articles::BustMultipleCachesWorker, args: [article_ids]) do
```

Comment:
We already have the articles here, so we don't need to re-fetch them 🙂 This also has the benefit of testing at a bit of a higher level:

```suggestion
    first = create(:article, tags: tag.name, published: true)
    second = create(:article, tags: tag.name, published: true)
    
    # Articles are enqueued in reverse chronological order
    sidekiq_assert_enqueued_with(job: Articles::BustMultipleCachesWorker, args: [[second.id, first.id]]) do
```

Ideally, we wouldn't have to assert on the _order_ of the article ids, just that it would be called with them, but that would require changes to `sidekiq_assert_enqueued_with` and I'm not about to heap that on you. 😄

Reply:
I fetched the articles to verify that the built articles were the ones getting fetched. But that was more of a debug requirement. On further thought, the extra fetch is unneccessary and less accurate. Thanks for the suggestion!
I could look into changing `sidekiq_assert_enqueued_with` if needed :D, would be exciting to work on a lower level.

Reply:
@squarebat 

> I fetched the articles to verify that the built articles were the ones getting fetched.

I recognize that was your intent, and I didn't call this out in my initial review, but since the query you were running in the spec was copy/pasted from the application code (or vice versa), it would certainly pass. It assumes the query returns those two built articles, but that wasn't explicitly verified in the spec, so it would most likely still pass if the query was incorrect.

Either way, thank you for updating it! :100:

> in this case checking for order is required, since we are looking to fetch the latest articles

This is a really good point. The only way to check for latest is to run them in reverse chronological order and limit results. It's a very pragmatic choice and I was being a bit pedantic there. 👍🏼

Reply:
Oh yeah, that makes sense, the test would definitely pass even if the query was faulty. Thanks for the insight. Will keep this in mind while writing future specs.